command_test.go

```diff
@@ -401,3 +403,35 @@ func TestCancel(t *testing.T) {
 	}
 	<-wait
 }
+
+type testEmitterWithError testing.T
+
+func (s *testEmitterWithError) Close() error {
+	return nil
+}
+
+func (s *testEmitterWithError) SetLength(_ uint64)                              {}
+func (s *testEmitterWithError) SetError(err interface{}, code cmdkit.ErrorType) {}
```

Comment:
We should record the error here and make sure it's set before the test ends.

Reply:
Done.